### ISSUE TITLE
"Undefined variable" error while importing type annotated variables from python file

### ISSUE THREAD

**Issue by JaPyR** [robocorp/robotframework-lsp#629]
**Describe the bug**
I am getting "Undefined variable" error while importing type annotated variable from python file.
I've noticed in 0.43.0 (also 0.43.2) but in 0.42.0 works fine.

**Example**
```python
# vars.py
from typing import Dict

VALUES: Dict[str, int] = {}
VALUES["foo"] = 1
VALUES["bar"] = 2

FOO: str = "foo"
BAR: int = 1
```
```robotframework
*** Settings ***
Variables    vars.py

*** Test Cases ***
Demo
    Log    ${VALUES}
    Log    ${FOO}
    Log    ${BAR}
```

**Screenshots**
![image](https://user-images.githubusercontent.com/7773301/160840768-25e17fe9-9453-48b0-a356-007ee7a7d969.png)

**Versions:**
 - OS: Win 10
 - Robot Framework Language Server Version: 0.42.0
 - Client Version:  1.65.2

**Comment by vtpeters** [comment]
I have the same experience, but also with variables that are built-in...

In a piece of code in one of my test cases I've constructed a List object containing files/documents.

With the FOR...IN statement I iterate through the List object and upload each file to a web page...
`FOR    ${document}    IN    @{documenten}
    ...
END`

After (automatically) upgrading to Robot Framework Language Server 0.43.x I noticed that the variable ${document} in the code had a wiggly line under it, indicating that there might be a syntax error. But the code runs fine, without error.

Screenshot after upgrading...
![image](https://user-images.githubusercontent.com/32060882/160885056-03373332-9dab-4499-bd04-6dddad80d95f.png)

Downgrading to Robot Framework Language Server 0.42.0 solved the issue. The wiggly lines were gone and the code still runs fine without errors.

OS: Win10
Microsoft VS Code: 1.65.2
Robot Framework 3.2.2
Robot Framework Language Server: 0.42.0

**Comment by pedrothaines** [comment]
I'm facing a very similar issue.

This might be related to #628.

**Comment by fabioz** [comment]
There really was an issue in the static analysis when dealing with annotated assigns.

Thanks for the report. I've just fixed that.

p.s.: for others facing issues with undefined variables, if it's not exactly the case of annotated assigns, please create different issues.
(or maybe take a look at: https://github.com/robocorp/robotframework-lsp/blob/master/robotframework-ls/docs/faq.md#how-to-deal-with-undefined-variables for how to deal with undefined variables in general).

**Comment by fabioz** [comment]
@vtpeters as a note, if your info is correct, you're using a `FOR` statement in `Robot Framework 3`... I believe `FOR` statements are only valid in `RF4`... if that's not your issue, please create a different issue.

**Comment by fabioz** [comment]
As a note, I've just published a pre-release (`0.43.4`) with the fix.
